From 56800d00a90af7ab58aeaafffa2eabc99cb29ec8 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Fri, 12 Dec 2025 16:21:16 +0000
Subject: [PATCH] Fix invocation of iptables if redirection fails

Fix the invocation of iptables/nftables if the redirection fails. As
os.Args[0] can still contain e.g. the full path of the symlink the
wrapper was invoked as, the leading path components must be stripped for
the actual iptables/nftables multi-call binary to know how to behave.
---
 main.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/main.go b/main.go
index caebb05..d604c87 100644
--- a/main.go
+++ b/main.go
@@ -41,6 +41,7 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+	"path"
 
 	"github.com/kubernetes-sigs/iptables-wrappers/internal/iptables"
 )
@@ -71,6 +72,7 @@ func main() {
 		// fake it, though this will probably also fail if they aren't root
 		binaryPath = iptables.XtablesPath(sbinPath, mode)
 		args = os.Args
+		args[0] = path.Base(args[0])
 	}
 
 	cmdIPTables := exec.CommandContext(ctx, binaryPath, args...)
-- 
2.51.2

